app/models/profile_host.rb

```diff
@@ -8,4 +8,10 @@ class ProfileHost < ApplicationRecord
 
   validates :profile, presence: true
   validates :host, presence: true, uniqueness: { scope: :profile }
+
+  after_destroy :destroy_orphaned_host
+
+  def destroy_orphaned_host
+    DeleteHost.perform_async(id: host.id) if host.profiles.empty?
+  end
```

Comment:
This is going to affect our migrations that delete profiles which haven't yet landed in QA or prod. We will want to make the migrations synchronous so that they can be rolled back if anything bad happens.